infrastructure/lib/dns-stack.ts

```diff
@@ -0,0 +1,67 @@
+import * as cdk from '@aws-cdk/core';
+import * as apigw from '@aws-cdk/aws-apigateway';
+import { Function } from '@aws-cdk/aws-lambda';
+import { ARecord, HostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+import * as route53Targets from '@aws-cdk/aws-route53-targets';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { SPADeploy } from 'cdk-spa-deploy';
+
+export interface DNSStackProps extends cdk.StackProps {
+    lambda: Function;
+}
+
+
+export class DNSStack extends cdk.Stack {
+    constructor(scope: cdk.Construct, id: string, props: DNSStackProps) {
+        super(scope, id, props);
+
+        const zone = HostedZone.fromLookup(this, `${id}-hz`, {
+            domainName: 'hobbycribbage.com',
+        });
+        // TODO it's safer to use `addResource` or `addMethod` or `addProxy` to point to this lambda
+        // than to have `proxy: true,`, because that's gonna just route every request to this lambda.
+        const restApi = new apigw.LambdaRestApi(this, 'cribbage-lambda-rest-api', {
+            restApiName: 'lambda-rest-api',
+            description: 'the bundled up golang server acting as a lambda',
+            handler: props.lambda,
+            proxy: true,
+            deployOptions: {
+                stageName: 'urlprefix',
+                // tracingEnabled: true,
+            },
+            endpointTypes: [apigw.EndpointType.EDGE],
+        });
+
+        const cert = new acm.Certificate(this, 'cribbage-lamdba-cert', {
+            domainName: '*.hobbycribbage.com',
+            validation: acm.CertificateValidation.fromDns(zone),
+        });
+        const subdomain = 'lambda.hobbycribbage.com';
+        restApi.addDomainName("domain_name", {
+            domainName: subdomain,
+            // securityPolicy: apigw.SecurityPolicy.TLS_1_2,
+            certificate: cert,
+            endpointType: apigw.EndpointType.EDGE,
+        })
+
+        new ARecord(this, `${id}-r53-a-record`, {
+            zone: zone,
+            recordName: subdomain,
+            comment: 'A comment on the ARecord',
+            target: RecordTarget.fromAlias(
+                new route53Targets.ApiGateway(
+                    restApi,
+                ),
+            ),
+        });
+
+        // https://github.com/cszczepaniak/oh-hell-scorecard/blob/7ea3b0ff4ce229d035bd5b100560399ac56be48c/infrastructure/lib/infrastructure-stack.ts#L8-L12
+        const zipFileName = 'spa-bundle';
```

Comment:
I guess I missed this in the previous PR, but just like our lambda zip file, we may want to put the git commit hash in the name here if we want to be able to easily deploy different commits (the commit could be an input to the GitHub action if we add a manual deploy option, even)